lib/AstOfLlbc.ml

```diff
@@ -652,11 +647,36 @@ let expression_of_scalar_value ({ C.int_ty; _ } as sv) : K.expr =
     let w = width_of_integer_type int_ty in
     K.(with_type (TInt w) (EConstant (constant_of_scalar_value sv)))
 
+let ascii_of_utf8_str (str : string) =
+  let get_uchar_list (str : string) =
+    let rec get_uchar_list acc idx =
+      try
+        let uchar = Uchar.utf_decode_uchar (String.get_utf_8_uchar str idx) in
+        let char_len = Uchar.utf_8_byte_length uchar in
+        get_uchar_list (uchar :: acc) (idx + char_len)
+      with
+      | Invalid_argument _ -> List.rev acc in
+    get_uchar_list [] 0 in
+  let uchar_list_to_ascii lst =
+    let to_str uchar =
+      if Uchar.is_char uchar then Uchar.to_char uchar |> Char.escaped
+      else Printf.sprintf "\\u{%x}" @@ Uchar.to_int uchar in
+    List.map to_str lst
+    |> String.concat "" in
+  get_uchar_list str
+  |> uchar_list_to_ascii
```

Comment:
this is also slightly inefficient in case you have large strings -- using a buffer and then `Printf.bprintf` would probably be more efficient, but I guess it's fine since we don't expect enormous strings